### ISSUE TITLE
Custom Column after aggregation creates wrong query and fails

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#12762]
**Describe the bug**
When adding a Custom Column with fields from aggregated data, then the query generated is mixing the original columns and the aggregated columns, which fails query with column not found.

**To Reproduce**
1. Custom question > Sample Dataset > Orders
2. Summarize "Sum of Subtotal" and "Sum of Total" by "CreatedAt:Year"
3. Custom Column `[Sum of Subtotal] + [Sum of Total]` as "MegaTotal"
![image](https://user-images.githubusercontent.com/1447303/85406731-815d0180-b562-11ea-8ba4-1d28dfbfa1ea.png)
4. Fails with error:
```
Column "source.SUBTOTAL" not found; SQL statement: CREATE FORCE VIEW PUBLIC._8 AS SELECT ("source"."sum" + "source"."sum_2") AS "MegaTotal", "source".SUBTOTAL AS SUBTOTAL, "source".TOTAL AS TOTAL, "source".CREATED_AT AS CREATED_AT, "source"."sum" AS "sum", "source"."sum_2" AS "sum_2", "source".CREATED_AT AS CREATED_AT_2 FROM ( SELECT PARSEDATETIME(YEAR(PUBLIC.ORDERS.CREATED_AT), 'yyyy') AS CREATED_AT, SUM(PUBLIC.ORDERS.SUBTOTAL) AS "sum", SUM(PUBLIC.ORDERS.TOTAL) AS "sum_2" FROM PUBLIC.ORDERS GROUP BY PARSEDATETIME(YEAR(PUBLIC.ORDERS.CREATED_AT), 'yyyy') ORDER BY 1 ) "source" [42122-197]
```
5. "View the SQL":
```SQL
SELECT "source"."CREATED_AT" AS "CREATED_AT",
       "source"."sum"        AS "sum",
       "source"."sum_2"      AS "sum_2",
       "source"."MegaTotal"  AS "MegaTotal"
FROM   (SELECT ( "source"."sum" + "source"."sum_2" ) AS "MegaTotal",
               "source"."SUBTOTAL"                   AS "SUBTOTAL",
               "source"."TOTAL"                      AS "TOTAL",
               "source"."CREATED_AT"                 AS "CREATED_AT",
               "source"."sum"                        AS "sum",
               "source"."sum_2"                      AS "sum_2",
               "source"."CREATED_AT"                 AS "CREATED_AT_2"
        FROM   (SELECT parsedatetime(year("PUBLIC"."ORDERS"."CREATED_AT"), 'yyyy') AS "CREATED_AT",
                       sum("PUBLIC"."ORDERS"."SUBTOTAL")                           AS "sum",
                       sum("PUBLIC"."ORDERS"."TOTAL")                              AS "sum_2"
                FROM   "PUBLIC"."ORDERS"
                GROUP  BY parsedatetime(year("PUBLIC"."ORDERS"."CREATED_AT"), 'yyyy')
                ORDER  BY parsedatetime(year("PUBLIC"."ORDERS"."CREATED_AT"), 'yyyy') ASC) "source") "source"
LIMIT  1048576  
```

Alternative way to reproduce similar problem with unneeded extra columns that causes the query to fail:
1. Simple question > Sample Dataset > Orders
2. Summarize by "Distinct values of User ID" and "Distinct values of Product ID", and save question as "Q1"
3. Custom question > Saved Questions > Q1
4. Add Custom Column `[Distinct values of Product ID] / [Distinct values of User ID]` as "test"
5. Visualize, the query fails with `Column "source.USER_ID" not found; ...`

**Information about your Metabase Installation:**
Metabase 0.35.4 and `master` on various backends and datasources - used to work in 0.35.3 and previous.

**Additional context**
I have a feeling that this might be related to #12507
https://discourse.metabase.com/t/error-on-use-summarized-column-on-custom-column/10623

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by mtthsbrr** [comment]
We also stumbled upon this when using the editor in 0.36.2. What we basically did:
- joining 4 tables
- creating 2 basic measures (distinct counts) in the summarize section (results are visualized)
- creating a custom measure (a / b) from the 2 basic measures in the next summarize section

Error: "ERROR: column source.foo does not exist Position:356"

Note: Using raw SQL works as expected

**Comment by aprada9** [comment]
Same here. We are currently on v0.36.6 and this happens when you create a custom column after summarizing, which was something working fine in previous versions (v.35)
Seems to be an error when building the query. 

It is kind of a blocker for us as most of our colleagues uses the custom columns so we will have to wait on previous Metabase versions until this is fixed.

**Comment by nemanjaglumac** [comment]
### NOTE
- a repro for this issue has been made (merged as https://github.com/metabase/metabase/commit/b87d4b5e1aa132395a93aca5193bae1fe0b34e15)
- you can run it in isolation as described [HERE](https://github.com/metabase/metabase/pull/13286) (replace `it.skip()` with `it.only()`)
- when working on a fix, enable (unskip) this test and make sure it passes
- merge it together with the fix

**Comment by camsaul** [comment]
Fixed by #13643
